src/useSkeleton.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import type { ReactNode } from 'react'
+import type { SkeletonProps } from './Skeleton'
+import { Skeleton } from './Skeleton'
+
+const emptyProps: SkeletonProps = {}
+
+/**
+ * Returns `withSkeleton` which conditionally returns a loading skeleton in place of contents
+ */
+export const useSkeleton = (loading: boolean, defaultProps = emptyProps) =>
+    React.useMemo(
+        () => ({
+            withSkeleton: <T extends ReactNode>(arg: T, props = emptyProps) =>
```

Comment:
TODO, since the returned object has only one key, remove the object wrapper and remove .useMemo()